File: stay/Drawer.js

const SvgDrawer = require('./SvgDrawer')
class Drawer {

    constructor(options) {
        this.svgDrawer = new SvgDrawer(options);
    }

    draw(data, target, themeName = 'light', infoOnly = false, highlight_atoms = []) {
        let canvas = null;
        if (typeof target === 'string' || target instanceof String) {
            canvas = document.getElementById(target);
        } else {
            canvas = target;
        }
        let svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
        svg.setAttribute('xmlns', 'http://www.w3.org/2000/svg')
        svg.setAttributeNS(null, 'viewBox', '0 0 ' + this.svgDrawer.opts.width + ' ' + this.svgDrawer.opts.height);
        svg.setAttributeNS(null, 'width', this.svgDrawer.opts.width + '');
        svg.setAttributeNS(null, 'height', this.svgDrawer.opts.height + '');
        this.svgDrawer.draw(data, svg, themeName, infoOnly, highlight_atoms);
        this.svgDrawer.svgWrapper.toCanvas(canvas, this.svgDrawer.opts.width, this.svgDrawer.opts.height);
    }

    getTotalOverlapScore() {
        return this.svgDrawer.getTotalOverlapScore();
    }

    getMolecularFormula() {
        this.svgDrawer.getMolecularFormula();
    }
}
module.exports = Drawer;

File: TRASH/чтото там/Drawer.py
import svgwrite

class SvgDrawer:
    def __init__(self, options):
        self.svgDrawer = svgwrite(options)
        pass

    def draw(self, data, svg, themeName='light', infoOnly=False, highlight_atoms=[]):
        pass

    def getTotalOverlapScore(self):
        pass

    def getMolecularFormula(self):
        pass

class Drawer:
    def __init__(self, options):
        self.svgDrawer = SvgDrawer(options)

    def draw(self, data, target, themeName='light', infoOnly=False, highlight_atoms=[]):
        svg = svgwrite.Drawing()
        svg.viewbox(0, 0, self.svgDrawer.opts.width, self.svgDrawer.opts.height)
        self.svgDrawer.draw(data, svg, themeName, infoOnly, highlight_atoms)
        # Implement logic to convert SVG to canvas

    def getTotalOverlapScore(self):
        return self.svgDrawer.getTotalOverlapScore()

    def getMolecularFormula(self):
        return self.svgDrawer.getMolecularFormula()

